src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -444,11 +476,15 @@ Copyright (c) .NET Foundation. All rights reserved.
           Returns="_TransitiveProjectReferences">
     <ItemGroup>
       <_TransitiveProjectDependencies Include="@(_ActiveTFMPackageDependencies->WithMetadataValue('TransitiveProjectReference', 'true'))" />
-
-      <!-- Get corresponding package definitions -->
-      <__TransitiveProjectDefinitions Include="@(PackageDefinitions)" Exclude="@(_TransitiveProjectDependencies)" />
-      <_TransitiveProjectDefinitions Include="@(PackageDefinitions)" Exclude="@(__TransitiveProjectDefinitions)" />
-      
+    </ItemGroup>
+    <!-- Get corresponding package definitions -->
+    <JoinItems Left="@(_TransitiveProjectDependencies)"
+               Right="@(PackageDefinitions)"
+               RightMetadata="*"
+                 >
```

Comment:
Nit: These aren't very long and I think it would be easier to follow with the attributes on one line. (Maybe not for the cases that use all of the options).